cmd/horcrux/cmd/config.go

```diff
@@ -218,13 +484,69 @@ type CosignerPeer struct {
 	P2PAddr string `json:"p2p-addr" yaml:"p2p-addr"`
 }
 
+func readKeyShare() (*signer.CosignerKey, error) {
+	// Read ID from local key share
+	var home string // In root.go we end up with our
+	if homeDir != "" {
+		home = homeDir
+	} else {
+		home, _ = homedir.Dir()
+		home = path.Join(home, ".horcrux")
+	}
+
+	if _, err := os.Stat(homeDir); !os.IsNotExist(err) {
+		return nil, fmt.Errorf("%s is not empty, check for existing configuration and clear path before trying again", homeDir)
+	}
+
+	file, err := os.Open(path.Join(home, "share.json"))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	bz, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+	var key signer.CosignerKey
+	if err := json.Unmarshal(bz, &key); err != nil {
+		return nil, err
+	}
+	return &key, nil
+}
+
 func validateCosignerPeers(peers []CosignerPeer) error {
-	// TODO: check IDs to ensure none are duplicated and
-	// that the key share which is configured for the local node
+	// Check IDs to make sure none are duplicated
+	if dupl := duplicatePeers(peers); len(dupl) != 0 {
+		return fmt.Errorf("found duplicate share IDs in args: %v", dupl)
+	}
+
+	// Check that the key share which is configured for the local node
 	// is the last remaining share
+	key, err := readKeyShare()
+	if err != nil {
+		return err
+	}
+	for _, peer := range config.CosignerPeers() {
+		if peer.ID == key.ID {
+			return fmt.Errorf("peer with share ID %v cannot be added", key.ID)
+		}
```

Comment:
Great safety check

Reply:
As far as I understand, I need the max number of peers to make this check, like if I have 3 peers, and IDs 1 and 2 are already present, the local node is obviously ID 3. I could use some pointers here, if you don't mind.

Reply:
I was thinking option 1: i.e. assume that the peers in the config are correct and that the node we are configuring is the missing ID

Reply:
Ok, here's what I did:
```bash
horcrux config peers set-shares [num-shares]
```
This adds a new field `shares` to the cosigner config, which sets the boundaries for adding new nodes/peers to the config.